src/tasks/phase2/utils.py

```diff
@@ -0,0 +1,344 @@
+    # plt.legend([])
+
+    # mng = plt.get_current_fig_manager()
+    # mng.window.state('zoomed')  # works fine on Windows!
+    plt.savefig("{0}{1}.png".format(output_image_path, "_".join(title.split())), dpi=500)
+    plt.show()
+    
+
+def plot_scree_test(eigen_values):
+    num_vars = len(eigen_values)
+
+    fig = plt.figure(figsize=(8, 5))
+    sing_vals = np.arange(num_vars) + 1
+    plt.plot(sing_vals, eigen_values, 'ro-', linewidth=2)
+    plt.title('Scree Plot')
+    plt.xlabel('K latent semantic')
+    plt.ylabel('Eigenvalue')
+
+    leg = plt.legend(['Eigenvalues from SVD'], loc='best', borderpad=0.3,
+                     shadow=False, prop=matplotlib.font_manager.FontProperties(size='small'),
+                     markerscale=0.4)
+    leg.get_frame().set_alpha(0.4)
+    #leg.draggable()
+    plt.show()
+
+def visualize_feature_semantics(latent_semantics, image_list, k, title = "Results", data_in_latent_space = None):
+    """
+       For Extra Credit
+       Author: Vibhu Varshney
+
+       :param latent_semantics: This is the latent semantic matric of shape(k, n) where k rows are the latent sematics and
+       n columns can be data objects or features based on what you are visualizing
+       :param image_list:
+       :param k: Keep it below 25 for better visualization
+       :param top_x: Number of top 'x' data objects or features that you want to display for each latent semantic. Keep it
+       below 10 for better visualization
+       :param k:
+
+       :return:
+       """
+    output_image_path = str(Path(os.getcwd()).parent) + "/Data/Output/"
+    # if (k > 15):
+    #     k = 15
+    #     latent_semantics = latent_semantics[:k]
+
+    m_sqrt = sqrt(k)
+    row = round(m_sqrt)
+    col = round(m_sqrt)
+    if (round(m_sqrt) < m_sqrt):
+        col = row + 1
+    sorted_tuples_images = []
+
+    for each_semantic in latent_semantics:
+        closest_image = get_top_m_tuples_by_similarity_score(data_in_latent_space,
+                                                             each_semantic, image_list,
+                                                             1, "cosine")
+        # pprint.pprint(closest_image)
+        sorted_tuples_images.append(closest_image[0])
+    index = 1
+    folder_path = get_image_directory()
+
+    for key, v in sorted_tuples_images:
+        plot_the_image_on_canvas(row, col, str(np.around(v, 4)), folder_path + "/" + key, index, figNum=200, showIndex=True)
+        index += 1
+    plt.suptitle(title)
+
+    # mng = plt.get_current_fig_manager()
+
+    # mng.window.state('zoomed')
+    plt.savefig("{0}{1}.png".format(output_image_path, "_".join(title.split())))
+    plt.show()
+
+
+def transform_cm(original_cm):
+    min_val = np.amin(original_cm)
+    print("orig ",original_cm.shape)
+    tranformed_cm = original_cm + abs(min_val)
+    print("trans ", tranformed_cm.shape)
+    # pprint.pprint(tranformed_cm)
+    # pprint.pprint(np.amin(tranformed_cm))
+    return tranformed_cm
+
```

Comment:
Most of the functions are not required cause we are going to use Django for frontend code.

Reply:
Removed.....just kept the scree plot function as we might need for finding 'K'